Source/Core/Common/CompatPatches.cpp

```diff
@@ -0,0 +1,275 @@
+    return false;
+  DWORD handle;
+  DWORD data_len = GetFileVersionInfoSizeW(path.c_str(), &handle);
+  if (!data_len)
+    return false;
+  std::vector<u8> block(data_len);
+  if (!GetFileVersionInfoW(path.c_str(), handle, data_len, block.data()))
+    return false;
+  void* buf;
+  UINT buf_len;
+  if (!VerQueryValueW(block.data(), LR"(\)", &buf, &buf_len))
+    return false;
+  auto info = static_cast<VS_FIXEDFILEINFO*>(buf);
+  *version = (static_cast<u64>(info->dwFileVersionMS) << 32) | info->dwFileVersionLS;
+  return true;
+}
+
+void CompatPatchesInstall(LdrWatcher* watcher)
+{
+  watcher->Install({{L"EZFRD64.dll", L"811EZFRD64.DLL"},
+                    [](const LdrDllLoadEvent& event) {
+                      // *EZFRD64 is incldued in software packages for cheapo third-party gamepads
+                      // (and gamepad adapters).
+                      // The module cannot handle its heap being above 4GB, which tends to happen
+                      // very often on modern Windows.
+                      // NOTE: The patch will always be applied, but it will only actually avoid the
+                      // crash if applied
+                      // before module initialization (i.e. called on the Ldr callout path).
+                      auto patcher = ImportPatcher(event.base_address);
+                      patcher.PatchIAT("kernel32.dll", "HeapCreate", HeapCreateLow4GB);
+                    }});
+  watcher->Install({{L"ucrtbase.dll"},
+                    [](const LdrDllLoadEvent& event) {
+                      // ucrtbase implements caching between fseek/fread, old versions have a bug
+                      // such that some reads return incorrect data. This causes noticable bugs
+                      // in dolphin since we use these APIs for reading game images.
+                      Version version;
+                      if (!GetModuleVersion(event.name.c_str(), &version))
+                        return;
+                      const u16 fixed_build = 10548;
+                      if (version.build >= fixed_build)
+                        return;
+                      const UcrtPatchInfo patches[] = {
+                          // 10.0.10240.16384 (th1.150709-1700)
+                          {0xF61ED, 0x6AE7B, 5},
+                          // 10.0.10240.16390 (th1_st1.150714-1601)
+                          {0xF5ED9, 0x6AE7B, 5},
+                          // 10.0.10137.0 (th1.150602-2238)
+                          {0xF8B5E, 0x63ED6, 2},
+                      };
+                      for (const auto& patch : patches)
+                      {
+                        if (ApplyUcrtPatch(event.name.c_str(), patch))
+                          return;
+                      }
+                      // If we reach here, the version is buggy (afaik) and patching failed
+                      auto msg = StringFromFormat(
+                          "You are running %S version %d.%d.%d.%d.\n"
```

Comment:
I wonder if this is going to cause problems for translators because they won't be able to put the version number before the file name... Do you think it would be acceptable to write this as `You are running %S %d.%d.%d.%d.` and have translators treat the whole `%S %d.%d.%d.%d` as a single unit?